include/dt-bindings/pinctrl/mchp-xec-pinctrl-common.h

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2021 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_MCHP_XEC_PINCTRL_COMMON_H_
+#define ZEPHYR_MCHP_XEC_PINCTRL_COMMON_H_
+
+/**
+ * @brief numerical IDs for IO ports
+ */
+
+/* Microchip XEC documentation expresses pin numbers in octal */
+#define MCHP_XEC_PORT_000_036	0 /* I/O Port for first group of 32 pins */
+#define MCHP_XEC_PORT_040_076	1
+#define MCHP_XEC_PORT_100_136	2
+#define MCHP_XEC_PORT_140_176	3
+#define MCHP_XEC_PORT_200_236	4
+#define MCHP_XEC_PORT_240_276	5
+
+#ifndef MCHP_XEC_PORTS_MAX
+#define MCHP_XEC_PORTS_MAX (MCHP_XEC_PORT_240_276 + 1)
+#endif
+
+/* GPIO is alternate function 0 */
+#define MCHP_XEC_PINMUX_FUNC_GPIO 0
+
+/** @brief helper macro to calcute pin port bit position from pin number */
+#define MCHP_XEC_PIN_POS(_pin) (_pin % 32)
+/**
+ * @brief helper macro to encode an IO port pin in a numerical format
+ */
+#define MCHPXECPIN(_port, _pin) \
+	(_port << 5 | _pin)
+
+#define MCHP_XEC_PIN_000	MCHPXECPIN(MCHP_XEC_PORT_000_036, 0)
```

Comment:
yuck, please delete these defines and just use the MCHPXECPIN macro as you need

Reply:
@jackrosenthal Hi, do you have a recommendation for pinctrl implementation?  I chose STM32 as it was recommened by @galak and the HW is similar: pin multiplexer is in the same HW register as most other GPIO pin features. Other SoC's are expressing pin DT as yaml_name = <&muxhw value1 value2 value3 ...>;  Matching MCHP documentation makes it slightly more difficult: pin numbers are in octa, gpio ports are named by pin number ranges instead of a, b, c, ...  I could do:
i2c1_scl: i2c1_scl {
  pinmux = <&gpio_100_136, 0131, 1>;
  bias-pull-up;
  drive-open-drain;
};
The C code either must save the phandle for use or calculate the port number from the pin number and pass that to the pin configuration API.
What's your preference?